src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorToolEnhanced.java

```diff
@@ -0,0 +1,1402 @@
+    private static final String DEFAULT_DETECTOR_DESCRIPTION = "Detector generated by OpenSearch Assistant";
+
+    private static final int SUGGEST_API_TIMEOUT_SECONDS = 30;
+    private static final int DATE_FIELD_QUERY_TIMEOUT_SECONDS = 10;
+    private static final int DATE_FIELD_SELECTION_TIMEOUT_SECONDS = 15;
+    private static final String DATE_FIELD_LOOKBACK_PERIOD = "now-30d";
+
+    private String name = TYPE;
+    private String description = DEFAULT_DESCRIPTION;
+    private String version;
+    private Client client;
+    private AnomalyDetectionNodeClient adClient;
+    private String modelId;
+    private ModelType modelType;
+    private String contextPrompt;
+    private Map<String, Object> attributes;
+
+    enum ModelType {
+        CLAUDE,
+        OPENAI;
+
+        public static ModelType from(String value) {
+            return valueOf(value.toUpperCase(Locale.ROOT));
+        }
+
+    }
+
+    /**
+     *
+     * @param client the OpenSearch transport client
+     * @param modelId the model ID of LLM
+     * @param modelType the model type (CLAUDE or OPENAI)
+     * @param contextPrompt custom prompt (if empty, loads from default file)
+     * @param namedWriteableRegistry the named writeable registry
+     */
+    public CreateAnomalyDetectorToolEnhanced(
+        Client client,
+        String modelId,
+        String modelType,
+        String contextPrompt,
+        NamedWriteableRegistry namedWriteableRegistry
+    ) {
+        this.client = client;
+        this.adClient = new AnomalyDetectionNodeClient(client, namedWriteableRegistry);
+        this.modelId = modelId;
+        if (!ModelType.OPENAI.toString().equalsIgnoreCase(modelType) && !ModelType.CLAUDE.toString().equalsIgnoreCase(modelType)) {
+            throw new IllegalArgumentException("Unsupported model_type: " + modelType);
+        }
+        this.modelType = ModelType.from(modelType);
+
+        if (contextPrompt.isEmpty()) {
+            this.contextPrompt = DEFAULT_PROMPT_DICT.getOrDefault(this.modelType.toString(), "");
+        } else {
+            this.contextPrompt = contextPrompt;
+        }
+
+        if (this.contextPrompt == null || this.contextPrompt.trim().isEmpty()) {
+            throw new IllegalArgumentException("Configuration error: detector creation prompt not found");
+        }
+    }
+
+    /**
+     * Creates anomaly detectors for specified indices with automatic validation and error handling.
+     * 
+     * @param parameters Map containing "input" with JSON array of index names
+     * @param listener ActionListener to receive results as JSON string
+     */
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        if (parameters.containsKey("input")) {
+            String inputStr = parameters.get("input");
+            if (inputStr != null && inputStr.trim().startsWith("[")) {
+                parameters.put("input", "{\"indices\": " + inputStr + "}");
```

Comment:
Should we sanitize user input? Is it possible we may have prompt injection from malicious users?